ENH Fix AttributeError of compound kernel of Gaussian processes (#22223)
scikit-learn/scikit-learn: doc/whats_new/v1.1.rst

  F-statistic).
  :pr:`17819` by :user:`Juan Carlos Alfaro Jiménez <alfaro96>`.

:mod:`sklearn.gaussian_process`
...............................

- |Fix| Raise better error if `CompoundKernel is passed to 

>   `GaussianProcessClassifier` as kernel .

Reviewer: So the class is linked correctly here:

```suggestion
  :class:`gaussian_process.GaussianProcessClassifier` as kernel .
```

> - |Fix| Raise better error if `CompoundKernel is passed to 
>   :class:`gaussian_process.GaussianProcessClassifier` as kernel .
>   :pr:`22223` by :user:`MarcoM <marcozzxx810>`.

Reviewer 2: ```suggestion
- |Fix| :class:`gaussian_process.GaussianProcessClassifier` raises
  a more informative error if `kernel=CompoundKernel` is passed.
  :pr:`22223` by :user:`MarcoM <marcozzxx810>`.
```
